lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/openturns/FunctionalChaosValidation.hxx

```diff
@@ -0,0 +1,83 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief The validation of a chaos expansion
+ *
+ *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_FUNCTIONALCHAOSVALIDATION_HXX
+#define OPENTURNS_FUNCTIONALCHAOSVALIDATION_HXX
+
+#include "openturns/MetaModelValidation.hxx"
+#include "openturns/FunctionalChaosResult.hxx"
+
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class FunctionalChaosValidation
+ *
+ * The validation of a chaos expansion
+ */
+
+class OT_API FunctionalChaosValidation
+  : public PersistentObject
+{
+  CLASSNAME
+
+public:
+  typedef Collection<Function>           FunctionCollection;
+
+  /** Default constructor */
+  FunctionalChaosValidation();
+
+  /** Parameter constructor */
+  FunctionalChaosValidation(const FunctionalChaosResult & chaosResult);
+
+  /** Virtual constructor */
+  FunctionalChaosValidation * clone() const override;
+
+  /** String converter */
+  String __repr__() const override;
+
+  /** Result accessor */
+  FunctionalChaosResult getResult() const;
+  
+  /** Compute LOO error */
+  Point computeLeaveOneOutError() const; 
+  
+  /** Compute LOO error */
+  Point computeKFoldError(const UnsignedInteger & k = 5) const; 
```

Comment:
Don't use hard-coded constants except when they have a clear meaning, e.g. acting like a flag. Prefer an entry in ResourceMap for this.

Reply:
Fixed at 2f4afd8d5b8658e55821b2abc2286c2c6d9e9686